contract/src/lib.rs

```diff
@@ -290,10 +293,32 @@ impl MpcContract {
     }
 
     #[allow(unused_variables)]
-    pub fn sign(&mut self, payload: [u8; 32]) -> [u8; 32] {
-        near_sdk::env::random_seed_array()
+    pub fn sign(&mut self, payload: [u8; 32]) -> Promise {
+        let hash = env::sha256(&payload);
+        self.pending_requests.insert(&hash, &None);
+        Self::ext(env::current_account_id()).sign_helper(hash)
```

Comment:
Note that this logic doesn't happen duplicate request. This makes it a bit easier for testing because we can reuse the same payload